artman/cli/converter.py

```diff
@@ -0,0 +1,297 @@
+    return result
+
+
+def _compute_artifacts(legacy_config, legacy_common):
+    result = []
+    LANGS = ['java', 'python', 'php', 'ruby', 'go', 'csharp', 'nodejs']
+
+    # Compute artifacts section
+    if legacy_common.get('git_repos'):
+        common_git_repos = legacy_common.get('git_repos')
+    else:
+        common_git_repos = []
+
+    for lang in LANGS:
+        if lang not in legacy_config:
+            continue
+        legacy_artifact_config = legacy_config[lang]
+        if legacy_artifact_config:
+            artifact = Artifact()
+            # Name the artifact as `{lang}_gapic`.
+            artifact.name = '%s_gapic' % lang
+            artifact.language = Artifact.Language.Value(lang.upper())
+
+            if 'release_level' in legacy_artifact_config:
+                artifact.release_level = Artifact.ReleaseLevel.Value(
+                    legacy_artifact_config.get('release_level').upper())
+
+            artifact.type = _compute_artifact_type(legacy_common)
+
+            # Compute package version.
+            if 'generated_package_version' in legacy_artifact_config:
+                legacy_package_version = legacy_artifact_config.get(
+                    'generated_package_version')
+                if 'lower' in legacy_package_version:
+                    artifact.package_version.grpc_dep_lower_bound = (
+                        legacy_package_version.get('lower'))
+                if 'upper' in legacy_package_version:
+                    artifact.package_version.grpc_dep_upper_bound = (
+                        legacy_package_version.get('upper'))
+
+            # Compute publishing targets
+            if 'git_repos' in legacy_artifact_config:
+                artifact.publish_targets.extend(_compute_publish_targets(
+                    legacy_artifact_config.get('git_repos'), common_git_repos))
+            result.append(artifact)
+    return result
+
+
+def _compute_artifact_type(legacy_common):
+    if legacy_common.get('packaging', '') == 'google-cloud':
+        return Artifact.GAPIC_ONLY
+    elif legacy_common.get('package_type', '') == 'grpc_common':
+        return Artifact.GRPC_COMMON
+
+    return Artifact.GAPIC
+
+
+def _compute_publish_targets(git_repos, common_git_repos):
+    result = []
+    for k, v in git_repos.items():
+        location = (common_git_repos[k]['location']
+                    if k in common_git_repos.keys() else v['location'])
+        target = Artifact.PublishTarget()
+        target.name = k
+        target.location = location
+        target.type = Artifact.PublishTarget.GITHUB
+
+        if 'paths' in v:
+            target.directory_mappings.extend(
+                _compute_directory_mappings(v['paths']))
+        result.append(target)
+    return result
+
+
+def _compute_directory_mappings(paths):
+    result = []
+    for path in paths:
+        mapping = Artifact.PublishTarget.DirectoryMapping()
```

Comment:
In general in places like this, you should compute a dictionary first and then assign it to the pb2 at the end.